devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -57,51 +71,49 @@ private void registerTasks(Project project) {
+                        // Quarkus test configuration task which should be executed before any Quarkus test
+                        t.dependsOn(quarkusTestConfig);
+                        // also make each task use the JUnit platform since it's the only supported test environment
+                        t.useJUnitPlatform();
+                    });
                 });
-
-        Task buildNative = tasks.create("buildNative", QuarkusNative.class);
-
-        // set up the source set for the testNative
-        JavaPluginConvention javaPlugin = project.getConvention().findPlugin(JavaPluginConvention.class);
-        if (javaPlugin != null) {
-            buildNative.dependsOn(tasks.getByName(BasePlugin.ASSEMBLE_TASK_NAME));
-
-            SourceSetContainer sourceSets = javaPlugin.getSourceSets();
-            SourceSet nativeTestSourceSet = sourceSets.create("native-test"); // this name has to be the same as the directory in which the tests reside
-            SourceSetOutput mainSourceSetOutput = sourceSets.getByName("main").getOutput();
-            SourceSetOutput testSourceSetOutput = sourceSets.getByName("test").getOutput();
-            nativeTestSourceSet.setCompileClasspath(
-                    nativeTestSourceSet.getCompileClasspath().plus(mainSourceSetOutput).plus(testSourceSetOutput));
-            nativeTestSourceSet.setRuntimeClasspath(
-                    nativeTestSourceSet.getRuntimeClasspath().plus(mainSourceSetOutput).plus(testSourceSetOutput));
-
-            // create a custom configuration to be used for the dependencies of the testNative task
-            ConfigurationContainer configurations = project.getConfigurations();
-            configurations.maybeCreate("nativeTestImplementation").extendsFrom(configurations.findByName("implementation"));
-            configurations.maybeCreate("nativeTestRuntimeOnly").extendsFrom(configurations.findByName("runtimeOnly"));
-
-            Task testNative = tasks.create("testNative", QuarkusTestNative.class).dependsOn(buildNative);
-            testNative.setShouldRunAfter(Collections.singletonList(tasks.findByName("test")));
-
-            tasks.getByName("check").dependsOn(testNative);
-        }
-
-        final QuarkusTestConfig quarkusTestConfig = tasks.create("quarkusTestConfig", QuarkusTestConfig.class);
-        tasks.withType(Test.class).forEach(t -> {
-            // Quarkus test configuration task which should be executed before any Quarkus test
-            t.dependsOn(quarkusTestConfig);
-            // also make each task use the JUnit platform since it's the only supported test environment
-            t.useJUnitPlatform();
-        });
     }
 
     private void verifyGradleVersion() {
-        if (GradleVersion.current().compareTo(GradleVersion.version("5.0")) < 0) {
-            throw new GradleException("Quarkus plugin requires Gradle 5.0 or later. Current version is: " +
+        if (GradleVersion.current().compareTo(GradleVersion.version("6.0")) < 0) {
```

Comment:
So we will require Gradle 6.x from now on?

This is something we definitely need to note. 
Also, have you tried generating a gradle project with `1.0.x.y` and tested if just bumping the gradle version will work with this PR?

Reply:
if it’s the case, would be good to know why we require gradle 6 ...as gradle users aren't agressive at updating afaik :)

Reply:
That's why I added the `noteworthy-feature` label :) 

Since I had to change the artifactId, 1.0.x projects will no longer work (the `useModule("quakus-gradle-plugin")` would fail), but I am planning on putting these migration steps in the migration guide: https://github.com/quarkusio/quarkus/wiki/Migration-Guide

Reply:
> if it’s the case, would be good to know why we require gradle 6 ...as gradle users aren't agressive at updating afaik :)

TBH we are not using any specific Gradle 6.x feature, I'll have a.look

Reply:
wait - why did you *have* to change the artifactid ?  

on gradle 5 vs 6 then I think better to stay conservative here and support 5+ (actually think we know we need higher than just 5.0, to make the enforced platform feature work)

Reply:
@maxandersen  as I explained above, changing the artifactId allows us to remove this snippet from the settings.gradle: 

```groovy
    resolutionStrategy {
        eachPlugin {
            if (requested.id.id == 'io.quarkus') {
                useModule("io.quarkus:quarkus-gradle-plugin:1.0.1.Final")
            }
        }
    }
```

I just tested, the plugin still works in Gradle 5, it's just that generated projects should target 6.x from now on. If you still need this plugin running on Gradle 5, some (minor) adjustments to the build.gradle and settings.gradle will be necessary (hint: it simplifies it a lot)

I'll make sure to put the necessary migration steps in the Migration guide for Gradle 5.x users.

Reply:
Also, since the plugin is no longer deployed to Maven central, `useModule` will not work

Reply:
I'm all for simplification :) 

Can we update the generated code part in this PR too ? just so that does not get forgotten...

I for one would like to see the actual differences incl. how gradle 5 and 6 usages will differ since then generation *wont* simple ;)

Reply:
I have updated `build.gradle` and `settings.gradle` in this PR already, feel free to compare its contents with the one produced by 1.0.1.Final. I'm sure you'll like it :)